subprojects/core-api/src/main/java/org/gradle/api/attributes/AttributeContainer.java

```diff
@@ -83,7 +83,7 @@ public interface AttributeContainer extends HasAttributes {
      * @return the attribute value, or null if not found
      */
     @Nullable
-    <T> T getAttribute(Attribute<T> key);
+    <T> T getAttribute(@Nullable Attribute<T> key);
```

Comment:
💭 Santa Tracker supplies `null` keys.  I've updated the interface and all impls to make this clear.